src/Farmer/Types.fs

```diff
@@ -199,6 +199,10 @@ type LinkedResource =
         | Unmanaged resId -> resId
     member this.Name = this.ResourceId.Name
 
+type DomainConfig =
+    | AppServiceDomain of string
```

Comment:
should this take the SSL Type as well to allow domains without SSL?

e.g.
```fsharp
type CertificateOptions = 
    | AppManagedCertificate of SslState
    | CustomCertificate of thumbprint: ArmExpression
    
type DomainConfig = 
    | SecuredDomain of domain:string * cert: CertificateOptions
    | InsecureDomain of domain:string
    | NoDomain
```
You'll probably need to move SslState into this file to enable this.